src/parser/de.rs

```diff
@@ -0,0 +1,309 @@
+use std::collections::HashMap;
+
+use serde::de::{DeserializeOwned, Error};
+use serde_json::{Map, Value};
+
+use crate::async_graphql_hyper::GraphQLRequestLike;
+
+type PosValHolder = HashMap<String, Vec<(String, String)>>;
+
+#[derive(Debug, serde::Deserialize, Default)]
+pub struct Parser {
+  root: Option<String>,
+  arguments: Option<String>,
```

Comment:
```suggestion
  arguments: HashMap<String, Vec<String>>,
```
Shouldn't arguments should be a hashmap.

Reply:
In the struct it just stores arguments string to later be parsed to hashmap check https://github.com/ssddOnTop/tailcall/blob/883118025e83f5ddca9ae4a9abc1ff5490414d13/src/parser/de.rs#L101

Reply:
Storing the map in struct will just consume more memory when we know that we only need the funal `Value` and it's going to be parsed only once